lib/publiccloud/gcp_client.pm

```diff
@@ -71,7 +72,27 @@ sub get_next_vault_role {
 sub create_credentials_file {
     my ($self) = @_;
     my $credentials_file;
+    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
+        my $json = get_credentials();
+        # Need to remove quotes, convert \n to \\n and / to \/
+        assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\\"//g' | sed 's/\\\n/\\\\\\\n/g' | sed 's/\\//\\\\\//g')});
+        #assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\"//g' | sed 's/\\n/\\\\n/g'      | sed 's/\//\\\//g')});
+
+        $self->private_key_id(get_required_var('PUBLIC_CLOUD_PRIVATE_KEY_ID'));
```

Comment:
Shouldn't PUBLIC_CLOUD_PRIVATE_KEY_ID be part of the `gce.json` file instead of an openQA variable? Like this we scatter the credentials handling between a credentials file and an openQA variable, thus it is on two different locations. Is there an advantage or necessity of doing so?

Reply:
Of course, and even all the other fields. But I just needed something to get the test running.